source/common/http/alternate_protocols_cache_impl.h

```diff
@@ -48,12 +49,16 @@ class AlternateProtocolsCacheImpl : public AlternateProtocolsCache {
   // Time source used to check expiration of entries.
   TimeSource& time_source_;
 
-  // Map from hostname to list of alternate protocols.
-  // TODO(RyanTheOptimist): Add a limit to the size of this map and evict based on usage.
-  std::map<Origin, std::vector<AlternateProtocol>> protocols_;
+  using ListType = std::list<std::pair<Origin, std::vector<AlternateProtocol>>>;
```

Comment:
I feel that `ListType` is too general. Can we make the alias sort of represent what's underneath?

Reply:
Yes, good point. This reminds me that instead of using std::pair, it would also be better to use a struct with named members. Both are done. Thanks!